### ISSUE TITLE
DropdownMenu accessibility: when expanded, only the dropdown items should be focusable

### ISSUE THREAD

**Issue by peter-adfx** [flutter/flutter#134297]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Turn on TalkBack
Open the dropdown menu
Swipe through the accessible elements
Note they are not limited to the items in the list, and even include completely obscured elements

### Expected results

When you expand the dropdown menu, accessibility focus should be limited to just the items in that menu, allowing screen readers to select from the list without interference from other elements on the screen.

In the code below, the "hello" text should not receive accessibility focus while the menu is expanded.

### Actual results

The items in the list are shown and added to the accessibility tree, but all other nodes remain. Typically we use the BlockSemantics widget to hide elements that are drawn before objects like this.

The "hello" text is highlighted by TalkBack.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  final TextEditingController intController = TextEditingController();

  MyApp();

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    final intEntries = <DropdownMenuEntry<int>>[];
    for (var i = 0; i < 5; i++) {
      intEntries.add(DropdownMenuEntry<int>(value: i, label: i.toString()));
    }

    return MaterialApp(
        title: 'Flutter Demo',
        home: Scaffold(
          body: SafeArea(
            child: Center(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Semantics(
                    button: true,
                    child: DropdownMenu<int>(
                      initialSelection: 1,
                      controller: intController,
                      dropdownMenuEntries: intEntries,
                    ),
                  ),
                  Text('hello'),
                ],
              ),
            ),
          ),
        ));
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots</summary>

![dropdown_menu_closed](https://github.com/flutter/flutter/assets/7657855/ec3e5cff-1b08-4360-8f5f-b54579eb7330)

![dropdown_menu_focus_outside](https://github.com/flutter/flutter/assets/7657855/fe1aad3d-4377-4d06-9f12-48ee65f78312)

  </details>

### Logs

No relevant logs

### Flutter Doctor output

Google internal details elided
  
    • Framework revision f019789565 (37 days ago), 2023-08-02T00:00:00.000
    • Engine revision 9304c0074d
    • Dart version afbaf4216f

**Comment by darshankawar** [comment]
Thanks for the report. Seeing same behavior as reported.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.13.2, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.13.2 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ff5b5b5fa6 (7 days ago), 2023-08-24 08:12:28 -0500
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.14.0-14.0.pre.93, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.14.0-14.0.pre.93 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ef9befc9da (11 hours ago), 2023-09-03 11:23:20 -0700
    • Engine revision fbc6f4a540
    • Dart version 3.2.0 (build 3.2.0-131.0.dev)
    • DevTools version 2.27.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by QuncCccccc** [comment]
It seems that the `MenuAnchor` shows the same behavior(When the icon button(anchor) is focused, the next focused thing is "hello", then the list items). @gspencergoog @LongCatIsLooong You might know more about the focus system, so just want to check if you have any ideas about the fix?

<details>
<summary> MenuAnchor code sample </summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {

  MyApp({super.key});
  final TextEditingController intController = TextEditingController();

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    final List<DropdownMenuEntry<int>> intEntries = <DropdownMenuEntry<int>>[];
    for (int i = 0; i < 5; i++) {
      intEntries.add(DropdownMenuEntry<int>(value: i, label: i.toString()));
    }

    return MaterialApp(
        title: 'Flutter Demo',
        home: Scaffold(
          body: SafeArea(
            child: Center(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  BlockSemantics(
                    child: MenuAnchor(
                      builder: (BuildContext context, MenuController controller, Widget? child) {
                        return IconButton(
                          onPressed: () {
                            controller.open();
                          },
                          icon: const Icon(Icons.search));
                      },
                      menuChildren: List<Widget>.generate(5, (int index)
                        => MenuItemButton(
                          child: Text('Item $index'),
                          onPressed: () {},
                        )
                      )
                    ),
                  ),
                  // DropdownMenu<int>(
                  //   initialSelection: 1,
                  //   controller: intController,
                  //   dropdownMenuEntries: intEntries,
                  // ),
                  const Text('hello'),
                ],
              ),
            ),
          ),
        ));
  }
}
```

</details>

<details>
<summary> video example </summary>

https://github.com/flutter/flutter/assets/36861262/c7490b42-554a-492b-a025-65e999445af3


</details>

**Comment by zanderso** [comment]
b/364794952

**Comment by QuncCccccc** [comment]
Will work on this soon.
